### ISSUE TITLE
[checkoutlinesufo] Do not check hash when in -w mode

### ISSUE THREAD

**Issue by frankrolf** [adobe-type-tools/afdko#283]
For a UFO file which has previously been edited with `checkoutlinesufo`, the line
`checkoutlinesufo -e -g N -w font.ufo/`
results in 
```
Skipped 1 of 1 glyphs.
Done with font
```

In my opinion, this is a logical mistake.
If the `-w` flag is explicitly set, overlaps should always be removed (and, perhaps., the background cleared?)

**Comment by miguelsousa** [comment]
@frankrolf It's not clear to me what the logical mistake is. Can you give an example step by step?

**Comment by frankrolf** [comment]
The `-w` mode removes overlaps in the foreground layer. If I explicitly ask `checkoutlinesufo` to write changes to that layer, I don’t care about the state of the `processedGlyphs` layer. 
This example will fail (2nd operation skipped):

```
checkoutlinesufo -e font.ufo
checkoutlinesufo -e -g X -w font.ufo
```

**Comment by miguelsousa** [comment]
@frankrolf please check this again using the changes made in #692 and let me know if there's still a problem. What I gathered from our offline discussion was that this issue is about the outlines in the default layer not getting updated rather than a specific problem with the logging messages.

**Comment by miguelsousa** [comment]
@frankrolf if you plan to run `makeotf` as well, you'll need the changes from #694, otherwise you'll get  this kind of messages:
```
src glyph xxx is newer than processed layer. programName: CheckOutlines
```

**Comment by frankrolf** [comment]
I checked with installing the FDK@784c839, and can observe the same behavior as described above.
The foreground glyph is skipped, based on the fact that it hasn’t changed since a previous checkOutlines run.
I can work around this problem by running `checkoutlinesufo -e -g N -w font.ufo --all` – but is that intuitive? I’m not sure.

**Comment by miguelsousa** [comment]
Is it intuitive to have to use the option `--all` in subsequent runs if you want all the glyphs to be processed (i.e. none skipped) ? Yes, it is, assuming that you're aware of the tool's options — if you're not, there's `checkoutlinesufo -h` to the rescue.

The requirement to specify `--all` is not unique to your example. If you use the option `-w` in the first run (instead of the second), the subsequent runs will also skip glyphs — unless `--all` is also used.

In summary, in both of the cases listed below, glyphs will be skipped on the second time **checkoutlinesufo** is run. This happens because the tool saves glyph hashes by default. If you don't want that to happen you'll need to use the `--clear-hash-map` option.

**Case 1**
```
checkoutlinesufo -e font.ufo
checkoutlinesufo -e -w font.ufo
```

**Case 2**
```
checkoutlinesufo -e -w font.ufo
checkoutlinesufo -e font.ufo
```
